docs/memory-management.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Memory Management
+...
+
+# Memory Management
+
+Memory management for a model consists of multiple parts including:
+
+- **Weights** which make the model
+- **Input, Output, and Label tensors** for the model
+- **Variables** required to run the model like intermediate inputs and outputs, derivatives, and gradients.
+
+The management for the above 3 kinds of memory is done depending on the mode of execution of the model, namely `inference` or `training` mode. The memory management of these tensors is performed by the `Manager` inside `NNTrainer`.
+
+The memory for the `weights` are allocated at the time of initializing the model, where the `weights` are either initialized using the provided initializer or loaded from the saved model file. The memory allocated for `weights` is freed
+upon `destruction` of the model object containing the `weights`.
```

Comment:
Concatenate the line 15, 16 will be better

Reply:
I will do a new PR since this is merged.